.circleci/config.yml

```diff
@@ -0,0 +1,305 @@
+            - dialyzer-cache-v2-{{ .Branch }}-{{ .Revision }}
+            - dialyzer-cache-v2-{{ .Branch }}-
+            - dialyzer-cache-v2-
+      - run:
+          name: Update dialyzer PLT
+          command: make dialyzer-install
+      - save_cache:
+          key: dialyzer-cache-v2-{{ .Branch }}-{{ .Revision }}
+          paths:
+            - _build/default/rebar3_20.2.2_plt
+      - run: make dialyzer
+
+  linux_package:
+    <<: *container_config
+    steps:
+      - checkout
+      - run: id
+      - *build_package
+      - *test_package
+      - *store_package_artifacts
+      - persist_to_workspace:
+          root: *packages_workspace
+          paths:
+            - "*.tar.gz"
+
+  osx_package:
+    macos:
+      xcode: "9.0"
+    working_directory: ~/epoch
+    steps:
+      - checkout
+      - run:
+          name: Install required tools
+          command: |
+            brew update
+            brew install erlang
+      - *build_package
+      - *test_package
+      - *store_package_artifacts
+
+  deploy_integration:
+    <<: *container_config
+    environment:
+      - DEPLOY_ENV: integration
+    steps: *deploy_steps
+
+  deploy_dev1:
+    <<: *container_config
+    environment:
+      - DEPLOY_ENV: dev1
+    steps: *deploy_steps
+
+  deploy_dev2:
+    <<: *container_config
+    environment:
+      - DEPLOY_ENV: dev2
+    steps: *deploy_steps
+
+  deploy_uat:
+    <<: *container_config
+    environment:
+      - DEPLOY_ENV: uat
+    steps: *deploy_steps
+
+# CircleCI skips a job for a tag by default.
+# A job must have a filters tags section to run as a part of a tag push
+# and all its transitively dependent jobs must also have a filters tags section.
+workflows:
+  version: 2
+  build_and_test:
+    jobs:
+      - build:
+          requires: ~
+          filters:
+            branches:
+              ignore:
+                - env/dev1
+                - env/dev2
+            tags:
+              only: /^v.*$/
+
+      - test:
+          requires:
+            - build
+          filters:
+            branches:
+              ignore:
+                - env/dev1
+                - env/dev2
+            tags:
+              only: /^v.*$/
+
+      - eunit:
+          requires:
+            - build
+          filters:
+            branches:
+              ignore:
+                - env/dev1
+                - env/dev2
+            tags:
+              only: /^v.*$/
+
+      - dialyzer:
+          requires:
+            - build
+          filters:
+            branches:
+              ignore:
+                - env/dev1
+                - env/dev2
+            tags:
+              only: /^v.*$/
+
+      - linux_package:
+          filters:
+            branches:
+              only:
+                - master
+                - env/dev1
+                - env/dev2
+            tags:
+              only: /^v.*$/
+
+      - osx_package:
+          requires:
+            - test
+            - eunit
+            - dialyzer
```

Comment:
We don't need to run dialyzer on OsX if we already have run it on ubuntu.

Reply:
This actually do not run the dialyzer twice, it's just and requirement that dialyzer must success before the osx packaging can start.

You can see this workflow here: https://circleci.com/workflow-run/19dfd816-6265-409f-990f-7921b9a97a7a

If I remove this requirement, the workflow could produce an osx package which is not successfully analyzed by dialyzer. I consider this broken package.